src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1889,18 +1899,20 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                         return TreeAccepts(this, TypeTree, type.TypeTree, out schemaDifference, out schemaDifferenceType, exact, useLegacyDateTimeAccepts);
                     }
 
-                    accepts = (IsMultiSelectOptionSet() && TypeTree.GetPairs().First().Value.OptionSetInfo == type.OptionSetInfo) || type.Kind == DKind.Unknown;
+                    accepts = (IsMultiSelectOptionSet() && TypeTree.GetPairs().First().Value.OptionSetInfo == type.OptionSetInfo) || type.Kind == DKind.Unknown || type.Kind == DKind.Deferred;
                     break;
 
                 case DKind.Enum:
-                    accepts = (Kind != type.Kind && type.Kind == DKind.Unknown) ||
+                    accepts = (Kind != type.Kind && (type.Kind == DKind.Unknown || type.Kind == DKind.Deferred)) ||
                               (EnumSuperkind == type.EnumSuperkind && EnumTreeAccepts(ValueTree, type.ValueTree, exact));
                     break;
 
                 case DKind.Unknown:
                     accepts = type.Kind == DKind.Unknown;
                     break;
-
+                case DKind.Deferred:
+                    accepts = type.Kind == DKind.Deferred || type.Kind == DKind.Unknown;
```

Comment:
Just in case of future use cases, as we want Deferred type to be sub type of all the type. So it can accept Unknown as well.

Reply:
Do we have code coverage for this line? If not, can we add it? Otherwise should this be an error in the short term What code path _could_ ever hit this?

We certainly would not want to have Deferred appear in the argument type list of a TexlFunction. The branches of If/Switch must all be the same, how does Deferred work there?

Reply:
We have a test in src/tests/Microsoft.PowerFx.Core.Tests/TypeSystemTests/DTypeTests.cs/DeferredIsSubtypeOfAll().

For computing return type of If/Switch, it takes supertype of all possible branch and since deferred is subtype of all if any other supertype is present that becomes the return type.